editor/src/components/canvas/canvas-strategies/draw-to-insert-strategy.spec.browser2.tsx

```diff
@@ -458,6 +459,71 @@ describe('Inserting into absolute', () => {
       `),
     )
   })
+
+  it('when drag ends outside the canvas in insert mode, it is cancelled', async () => {
+    const renderResult = await renderTestEditorWithCode(inputCode, 'await-first-dom-report')
+
+    const targetElement = renderResult.renderedDOM.getByTestId('bbb')
+    const targetElementBounds = targetElement.getBoundingClientRect()
+    const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+    const startPoint = slightlyOffsetWindowPointBecauseVeryWeirdIssue({
+      x: targetElementBounds.x + 5,
+      y: targetElementBounds.y + 5,
+    })
+
+    const endPoint = slightlyOffsetWindowPointBecauseVeryWeirdIssue({
+      x: targetElementBounds.x + 1005,
+      y: targetElementBounds.y + 1005,
+    })
+
+    mouseMoveToPoint(canvasControlsLayer, startPoint)
+
+    await startInsertMode(renderResult.dispatch)
+
+    mouseDragFromPointToPointNoMouseDown(canvasControlsLayer, startPoint, endPoint)
+
+    await renderResult.getDispatchFollowUpActionsFinished()
+
+    // Check that the inserted element is a child of bbb
+    expect(getPrintedUiJsCode(renderResult.getEditorState())).toEqual(
+      makeTestProjectCodeWithSnippet(`
```

Comment:
if there's no expected changes, you should be able to reuse `inputCode` here

Reply:
fixed, thanks!